### DISCUSSION TITLE
Dev Meeting 230517

### DISCUSSION THREAD

**Discussion by ehhartmann** [graeter-group/kimmdy#115] [Ideas]
**Goal : Create project plan for releasing KIMMDY v2.0**

#### Requirements stated in 221212 Meeting:

> - Using KIMMDY/Release  
	- Partial charges should be good  
	- Espaloma parameters as new verions  
	- Publishing:  - Methods, - one use case / example like hexala for statistics and details , -one big one to show off, - maybe use ala-ala-dopa-ala-ala  
	- Documentation: - example, - auto generated technical documentation  
	- Features: ,- Espaloma -> next paper ,- Collagen -> next paper ,- MC w/ rejection + testing -> first release!  

#### Tasks
features:
- [x] finish KMC implementations **Eric, Kai**
- [x] calculate all necessary partial charges **Eric**
  - still missing but not necessary for now: terminal charges, crosslinks, some His species (maybe combine with extending radical QM dataset)
- [x] get partial charges into KIMMDY (validate) **Jannik**
- [x] finish relaxation schemes (slow growth and hydrogen placement **Eric**
- [x] implement/ improve choice of starting structure for relaxation  **Eric, hat predicted**
- [x] clean up config file (and config.py) **Jannik, Kai**
- [ ] write analysis tools for kimmdy **all**
   - improve trajectory concatenating tool to get meaningful trajectories based on KMC scheme
   - graph/vmd visualisation of population of radical positions in the molecule
   - graph/vmd visualisation of radical flux from each position
   - graph/vmd visualisation of most probable radical migration pathways
   - analysis of racemization
   - plot energy profile over simulation (of protein only)
- [x] HAT reaction plugin **Kai**

testing
- [x] fix tests
- [ ] extend tests
- [x] build examples from tests?

go through issues

documentation:
- tidy up **Jannik**
- add docstrings 
- clean up examples **Kai**

code review 

publishing:
- do simulations of toy systems
- analyse simulations
- write

**Comment by KRiedmiller** [comment]
* Reaction API: whats the difference between ReactionOutcome and ReactionResult and should there be any?

**Reply by jmbuhr** [reply]
One ReactionResult contains a list of potential ReactionOutcomes (which is a rate and a recipe)

**Reply by jmbuhr** [reply]
naming things is hard 🙈

**Reply by KRiedmiller** [reply]
But ReactionOutcome is itself already a list of reactions recipes, rates and framenumbers. We're just adding an empty dimension (in numpy terms) 
(also talking about the newest 'betterKMC branch')

**Comment by ehhartmann** [comment]
We have a little more than one week left before the meeting with Frauke. I would like to focus on some parts to have a working version until then (also, we could present kimmdy at the retreat). 
If possible, having Leif's parameters in kimmdy would be great. Also, some basic visualization/analysis tools and a working HAT plugin would add a lot. Do you think its possible to finish these features until then @jmbuhr @KRiedmiller ? Would you add/remove something?

**Reply by jmbuhr** [reply]
I want to finish some cleaning-up today and add more tests in the process so that I see what breaks when I add the parameters, but if this goes well we can move pretty fast.
I also added markdown support for the docs and an example of a quarto document that get's rendered to markdown, to then be picked up by the docs, so you can add visualizations to this easily, though this is still a bit flimsy and lately sphinx is ignoring the folder with the image outputs... But if I get this to work properly we can show off analysis tools very nicely.
